### PR TITLE
support for home.avdl RPCs

### PR DESCRIPTION
**Pull request by maxtaco**
- a simple HomeGet RPC for getting the home screen
- cache busting via gregor for this data being cached
- plumb through a POST skipping todo types
- a simple systest for this, but enough to get us started

### CODE REVIEW

File: go/client/commands_common.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ func GetCommands(cl *libcmdline.CommandLine, g *libkb.GlobalContext) []cli.Comma
 		NewCmdDumpPushNotifications(cl, g),
 		NewCmdEncrypt(cl, g),
 		NewCmdGit(cl, g),
+		NewCmdHome(cl, g),
```
Comment (mlsteele) [inline go/client/commands_common.go]: Should this be a dev only command, at least for now?

File: go/home/home.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	h.Lock()
+	defer h.Unlock()
+
+	if h.cache != nil {
+		diff := h.G().GetClock().Now().Sub(h.cache.cachedAt)
+		if diff < libkb.HomeCacheTimeout {
+			h.G().Log.CDebugf(ctx, "| hit cache (cached %s ago)", diff)
+			if markViewed {
+				h.G().Log.CDebugf(ctx, "| going to server to mark view, anyways")
+				tmp := h.markViewed(ctx)
```
Comment (mlsteele) [inline go/home/home.go]: name `tmp` `err`

File: go/systests/home_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	require.Equal(t, home.Version, 0, "on home version 0")
+	require.Equal(t, home.LastViewed, keybase1.Time(0), "never viewed before")
+	assertTodoPresent(t, home, keybase1.HomeScreenTodoType_FOLLOW)
+
+	iui := newSimpleIdentifyUI()
+	iui.confirmRes = keybase1.ConfirmResult{IdentityConfirmed: true, RemoteConfirmed: true, AutoConfirmed: true}
+	attachIdentifyUI(t, g, iui)
+	alice.track(wong.username)
+
+	wait := 10 * time.Millisecond
```
Comment (mlsteele) [inline go/systests/home_test.go]: Comment that the total wait is 630ms (I think). Is that a little short? Most of CI-is-slow tests wait for longer.

File: go/client/cmd_home.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
```
Comment (mlsteele) [inline go/client/cmd_home.go]: those were the days

Comment (maxtaco) [reply go/client/cmd_home.go]: :)

File: go/client/commands_common.go (hunk 1 above)
Comment (maxtaco) [reply go/client/commands_common.go]: how about a hidden command?

File: go/systests/home_test.go (hunk 3 above)
Comment (maxtaco) [reply go/systests/home_test.go]: makes sens

File: go/home/home.go (hunk 2 above)
Comment (maxtaco) [reply go/home/home.go]: Let's say `tmpErr` so I don't worry about it shadowing the `err` for the whole function

File: go/systests/home_test.go (hunk 3 above)
Comment (maxtaco) [reply go/systests/home_test.go]: OK, I'll make it 10 iters, which is about ~20s by the back of  my envelope

File: go/client/commands_common.go (hunk 1 above)
Comment (mlsteele) [reply go/client/commands_common.go]: yeah, that's sounds fine

File: go/home/home.go (hunk 2 above)
Comment (mlsteele) [reply go/home/home.go]: Ok if you like. Shadowing wouldn't be bad here. Also you could do `err = h.markViewed(ctx)` and that would re-assign without shadowing.

File: go/systests/home_test.go (hunk 3 above)
Comment (mlsteele) [reply go/systests/home_test.go]: Cool. `python: sum([10*(2**i) for i in range(10)]) / 1000. -> 10.23`

**Pull request merged**